content/en/blog/_posts/2024-04-17-kubernetes-1.30.md

```diff
@@ -0,0 +1,191 @@
+
+### Prevent unauthorized volume mode conversion during volume restore ([SIG Storage](https://github.com/kubernetes/community/tree/master/sig-storage))
+
+With the release of Kubernetes 1.30, the ability to inherently prevent unauthorized users from converting the volume mode of a PVC, when it is created from an existing VolumeSnapshot, moves to GA. Cluster administrators will now have to specifically grant permissions in Kubernetes to authorized users, like storage backup vendors, to allow them to perform such actions. 
+For more information, see [converting the volume mode of a Snapshot](/docs/concepts/storage/volume-snapshots/#convert-volume-mode).
+
+### Pod Scheduling Readiness ([SIG Scheduling](https://github.com/kubernetes/community/tree/master/sig-scheduling))
+
+Blurb for 3521 goes here.
+
+### Min domains in PodTopologySpread ([SIG Scheduling](https://github.com/kubernetes/community/tree/master/sig-scheduling))
+
+Blurb for 3022 goes here.
+
+### Go workspaces for k/k ([SIG Architecture](https://github.com/kubernetes/community/tree/master/sig-architecture))
+
+The Kubernetes repo now uses Go workspaces. This should not impact end users at all, but does have a impact for developers of downstream projects. Switching to workspaces caused some breaking changes in the flags to the various [k8s.io/code-generator](https://github.com/kubernetes/code-generator) tools. Downstream consumers should look at [`staging/src/k8s.io/code-generator/kube_codegen.sh`](https://github.com/kubernetes/code-generator/blob/master/kube_codegen.sh) to see the changes.
+
+For full details on the changes and reasons why Go workspaces was introduced, read [Using Go workspaces in Kubernetes](https://www.kubernetes.dev/blog/2024/03/19/go-workspaces-in-kubernetes/).
+
+## Improvements that graduated to beta in Kubernetes v1.30 {#graduations-to-beta}
+
+_This is a selection of some of the improvements that are now beta following the v1.30 release._
+
+### Node log query ([SIG Windows](https://github.com/kubernetes/community/tree/master/sig-windows))
+
+To help with debugging issues on nodes, Kubernetes v1.27 introduced a feature that allows viewing logs of services running on the node. To use the feature, ensure that the NodeLogQuery feature gate is enabled for that node, and that the kubelet configuration options enableSystemLogHandler and enableSystemLogQuery are both set to true. 
+
+On Linux the assumption is that service logs are available via journald. On Windows the assumption is that service logs are available in the application log provider. On both operating systems, logs are also available by reading files within /var/log/.
+[To read more about the feature, please click here](https://kubernetes.io/docs/concepts/cluster-administration/system-logs/#log-query)
+
+### CRD Validation Ratcheting ([SIG API Machinery](https://github.com/kubernetes/community/tree/master/sig-api-machinery))
+
+Blurb for 4008 goes here--subject to Docs Freeze
```

Comment:
```suggestion
You need to enable the `CRDValidationRatcheting`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to
use this behavior, which then applies to all CustomResourceDefinitions in your
cluster.

Provided you enabled the feature gate, Kubernetes implements _validation racheting_
for CustomResourceDefinitions. The API server is willing accept updates to resources that
are not valid after the update, provided that each part of the resource that failed to validate
was not changed by the update operation. In other words, any invalid part of the resource
that remains invalid must have already been wrong.
You cannot use this mechanism to update a valid resource so that it becomes invalid.

This feature allows authors of CRDs to confidently add new validations to the
OpenAPIV3 schema under certain conditions. Users can update to the new schema
safely without bumping the version of the object or breaking workflows.
```